### ISSUE TITLE
Images are not loaded inside relations

### ISSUE THREAD

**Issue by denisgoryaynov** [gatsby-uc/gatsby-source-strapi#308]
I have a content type called "Product Category" with a one to many relation to itself. A "Product Category" also has a one to many relation with "Product" content type.
When dealing with the following tree:
- Product Category "main"
   - Product Category "sub-main"
      - Product 1
      - Product 2
      
<img width="412" alt="Screenshot 2022-03-20 at 15 15 41" src="https://user-images.githubusercontent.com/16140813/159166789-6b06c4f6-b008-4e1d-9a92-ee5cc9da32b4.png">

Images for "Product" are not loaded, I can see the featured_image field inside graphql, but the localFile field is not available so I cannot use gatsby-image with it

I am using Strapi 4.2.0-beta.0 and gatsby-source-strapi 2.0.0-beta.0

Edit: at least in my case adding the following code after this line https://github.com/strapi/gatsby-source-strapi/blob/2a60320b16ae6520617d743158c433d384dbd155/src/download-media-files.js#L126 fixes the issue:

```
if (type === 'relation' && Array.isArray(value)) {
   for (const element of value) {
      await extractImages(element, ctx, attribute.target);
   }
}
```

**Comment by IanSpringer** [comment]
Hi,

Is there any update on this issue? This has been a huge blocker for me and my team.

**Comment by soupette** [comment]
@IanSpringer are you populating the images from you relations?

**Comment by IanSpringer** [comment]
@soupette I have populated the images in my relations, yes. I actually created an issue a few days back about this same thing thinking it was a problem with relations + dynamic zones. See here https://github.com/strapi/gatsby-source-strapi/issues/336

When adding an image to a collection type, and then pulling that collection item as a relation in our dynamic zone, we cannot query localFile on the images. This is a big problem as we rely on gatsby image sharp processing. Without this capability, we will take a big hit in our performance, or run the risk of spending lots of hours on dev time to try and hack a fix.
